client/hotrod-client-new/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/ChannelHandler.java

```diff
@@ -86,6 +87,8 @@ public ChannelHandler(Configuration configuration, String sniHostName, ExecutorS
                null, ClientAuth.NONE, null, false);
          this.watcher = null;
       }
+
+      configuration.metricRegistry().createGauge("connection.pool.size", "The total number of connections", channels::size, Map.of(), null);
```

Comment:
@wburns added this gauge about the number of connections in the new client.

Reply:
Sounds good

Reply:
The 
      totalRetriesMetric = metricsRegistry.createCounter("connection.pool.retries", "The total number of retries", Map.of(), null);
can be added to the OperationDispatcher now. That is the class whose sole job is to dispatch operations to the correct servers so it manages retries.

Reply:
Thanks! I missed that one :+1: 
Added!